azext_iot/iothub/params.py

```diff
@@ -246,3 +246,88 @@ def load_iothub_arguments(self, _):
             options_list=["--content-type", "--ct"],
             help="MIME Type of file.",
         )
+
+    with self.argument_context("iot edge devices") as context:
+        context.argument(
+            "devices",
+            options_list=["--device", "-d"],
+            nargs="+",
+            action="append",
+            help="Space-separated key=value pairs corresponding to properties of the edge device to create. "
+            "The following key values are supported: `id`, `deployment` (inline json or path to file), `hostname`, "
+            "`parent`, `edge_agent`, and `container_auth` (inline json or path to file). "
+            "--device can be used 1 or more times. Review help examples for parameter usage."
+        )
+        context.argument(
+            "clean",
+            options_list=["--clean", "-c"],
+            arg_type=get_three_state_flag(),
+            help="Deletes all devices in target hub before creating new devices.",
+        )
+        context.argument(
+            "visualize",
+            options_list=["--visualize", "--vis", "-v"],
+            arg_type=get_three_state_flag(),
+            help="Shows visualizations of devices and progress of various tasks "
+            "(device creation, setting parents, updating configs, etc).",
+        )
+        context.argument(
+            "config_file",
+            options_list=["--config-file", "--config", "--cfg"],
+            help="Path to devices configuration file. Sample configuration file: "
+            "https://github.com/Azure/azure-iot-cli-extension/tree/dev/docs/samples/sample_devices_config.yaml",
+        )
+        context.argument(
+            "device_auth_type",
+            arg_type=get_enum_type(
+                [DeviceAuthType.shared_private_key.value, DeviceAuthType.x509_thumbprint.value]
+            ),
+            options_list=["--device-auth-type", "--device-auth"],
+            help="Device to hub authorization mechanism.",
+        )
+        context.argument(
+            "default_edge_agent",
+            options_list=["--default-edge-agent", "--default-agent", "--dea"],
+            help="Default edge agent for created Edge devices if not specified individually.",
+        )
+        context.argument(
+            "device_config_template",
+            options_list=["--device-template", "--dt"],
```

Comment:
whats the difference between this and config file?

Reply:
This is the edge `config.toml` device configuration template used as a basis for each edge [device's config](https://learn.microsoft.com/en-us/azure/iot-edge/how-to-provision-single-device-linux-x509?view=iotedge-1.4&tabs=azure-portal%2Cubuntu#provision-the-device-with-its-cloud-identity):

See [device_config.toml](https://github.com/Azure/azure-iot-cli-extension/pull/591/files/db13535a20ae1d5f600b1fd6b67c127d79226cca#diff-7ca392c8d88a5cc372d64ce3e35b9ede99a5ef0f7951edd65f264e9bf4808cf1)